### ISSUE TITLE
[Question] Robotiq 2F-85 Gripper Deadlock, Arm Oscillation, and Link Dislocation with ImplicitActuator (Newton Engine @ 240Hz)

### ISSUE THREAD

**Issue by N3rv3rM1nd** [isaac-sim/IsaacLab#5083]
Hello Isaac Lab team,

Sorry for disturbing you, I truly appreciate the time and effort you take to look into these technical challenges.I am experiencing severe physics instability (deadlock and violent oscillation) when trying to close the official Robotiq 2F-85 gripper attached to a UR5e arm. I am running the simulation with the Newton solver at dt=1/240 to handle high-stiffness parameters.

When sending a close command (0.5 rad) to the gripper's active joint, it instantly deadlocks at around 0.008 rad. Meanwhile, the physical solver seems to generate massive opposing forces . This force propagates up the kinematic chain, causing the UR5e's wrist_2_joint to violently oscillate between -1.1 and -1.8 rad. In some attempts, the passive links of the gripper even visually dislocate from their pivot points.

I have tried various methods to control the finger_joint, but none of them have worked. My gripper USD file is sourced from the Isaac Sim 6.0 cloud asset library, and the only modifications I made were removing the root articulation node and disabling self-collisions among the gripper joints. My UR5e USD file is sourced from the Isaac Sim 5.1 cloud assets, which I have tested independently without any issues.

I noticed that the stiffness values configured in both the gripper and arm USD files are extremely high. I initialized my setup strictly following these default USD values. The detailed parameters are provided below.

Isaac Lab Version: [3.0.0]
Isaac Sim Version: [6.0.0]
OS: Ubuntu 24.04
GPU: RTX 4090

```import argparse
import torch
import warp as wp
from isaaclab.app import AppLauncher

parser = argparse.ArgumentParser(description="UR5e + 2F85 Streamlined")
AppLauncher.add_app_launcher_args(parser)
args_cli = parser.parse_args()
app_launcher = AppLauncher(args_cli)
simulation_app = app_launcher.app

import isaaclab.sim as sim_utils
from isaaclab.assets import ArticulationCfg, Articulation
from isaaclab.sim import SimulationCfg, SimulationContext
from isaaclab.actuators import ImplicitActuatorCfg
from isaaclab_newton.physics import NewtonCfg, MJWarpSolverCfg

robot_cfg = ArticulationCfg(
    prim_path="/World/Robot",
    spawn=sim_utils.UsdFileCfg(
        usd_path="/root/gpufree-data/workspace/IsaacLab/ur5e_isaac_v2_flattened.usd",
        articulation_props=sim_utils.ArticulationRootPropertiesCfg(enabled_self_collisions=False),
    ),
    init_state=ArticulationCfg.InitialStateCfg(
        pos=(0.0, 0.0, 0.0),
        joint_pos={
            "shoulder_pan_joint": 0.0,
            "shoulder_lift_joint": -1.57,
            "elbow_joint": 1.5708,
            "wrist_1_joint": -1.5708,
            "wrist_2_joint": -1.5708,
            "wrist_3_joint": 0.0,
            "finger_joint": 0.0,
            ".*_inner_finger_joint": 0.0,
            ".*_inner_finger_knuckle_joint": 0.0,
            "right_outer_knuckle_joint": 0.0,
        },
    ),
    actuators={
        "ur5e_shoulder_pan": ImplicitActuatorCfg(joint_names_expr=["shoulder_pan_joint"], stiffness=9400.50098, damping=0.378, armature=0.0),
        "ur5e_shoulder_lift": ImplicitActuatorCfg(joint_names_expr=["shoulder_lift_joint"], stiffness=10020.91992, damping=0.41168, armature=0.0),
        "ur5e_elbow": ImplicitActuatorCfg(joint_names_expr=["elbow_joint"], stiffness=10230.2207, damping=4.09288, armature=0.0),
        "ur5e_wrist_1": ImplicitActuatorCfg(joint_names_expr=["wrist_1_joint"], stiffness=3940.62891, damping=1.57851, armature=0.0),
        "ur5e_wrist_2": ImplicitActuatorCfg(joint_names_expr=["wrist_2_joint"], stiffness=3940.62891, damping=0.06109, armature=0.0),
        "ur5e_wrist_3": ImplicitActuatorCfg(joint_names_expr=["wrist_3_joint"], stiffness=1000.08087, damping=0.00432, armature=0.0),
        "gripper_active": ImplicitActuatorCfg(joint_names_expr=["finger_joint"], stiffness=30000.0, damping=0.0, armature=1.0),
        "gripper_passive": ImplicitActuatorCfg(joint_names_expr=[".*_inner_finger_joint", ".*_inner_finger_knuckle_joint", "right_outer_knuckle_joint"], stiffness=0.0, damping=0.0, armature=0.0),
    },
)

sim_cfg = SimulationCfg(dt=1/240, physics=NewtonCfg(solver_cfg=MJWarpSolverCfg()))
sim_context = SimulationContext(sim_cfg)
robot = Articulation(cfg=robot_cfg)

sim_context.reset()
robot.reset()

def get_tensor(data_array):
    return wp.to_torch(data_array).clone() if hasattr(data_array, "ptr") else data_array.clone()

arm_idx = [0, 1, 2, 3, 4, 5]
g_act_idx = [6]
num_joints = robot.num_joints
device = robot.device if hasattr(robot, 'device') else "cuda:0"
target_positions = torch.zeros((1, num_joints), device=device)

safe_arm_pose = torch.tensor([0.0, -1.5708, 1.5708, -1.5708, -1.5708, 0.0], device=device)
force_spawn_pos = get_tensor(robot.data.joint_pos)
force_spawn_pos[0, arm_idx] = safe_arm_pose
robot.write_joint_state_to_sim(force_spawn_pos, get_tensor(robot.data.joint_vel))
robot.reset()

downward_pose = torch.tensor([0.0, -1.0, 1.0, -1.5708, -1.5708, 0.0], device=device)

step_count = 0
while simulation_app.is_running():
    blend_factor = min(step_count / 800.0, 1.0)
    current_arm_target = (1.0 - blend_factor) * safe_arm_pose + blend_factor * downward_pose
    target_positions[0, arm_idx] = current_arm_target
    target_positions[0, g_act_idx[0]] = 0.5

    robot.set_joint_position_target(target_positions)
    robot.write_data_to_sim()
    sim_context.step()
    robot.update(dt=sim_context.get_physics_dt())
    step_count += 1

simulation_app.close()```


I am eager to hear your thoughts on this and am grateful for any guidance you can provide to help me resolve this deadlock. Thank you again for your time and for all the incredible work you do on Isaac Lab!　QWQ

**Comment by e-ndorfin** [comment]
hey! not part of the isaac lab team but would be great if you could wrap your code in ```python tags so that the team can easily run and replicate your error

also adding my own input, I'm currently working on a UR5 arm + 2F-85 gripper and have found these settings to work, unsure if they might be able to help you. I also tried initializing with the default USD values from the 2F-85 gripper


```python
            "ur5_shoulder": ImplicitActuatorCfg(
                joint_names_expr=[
                    "shoulder_pan_joint",
                    "shoulder_lift_joint",
                    "elbow_joint"
                ],
                effort_limit_sim=150.0,
                stiffness=1008.43994,
                damping=40.03376,
            ),
            "ur5_wrist": ImplicitActuatorCfg(
                joint_names_expr=["wrist_[1-3]_joint"],
                effort_limit_sim=28.0,
                stiffness=379.82324,
                damping=10.51,
            ),
            "gripper": ImplicitActuatorCfg(
                joint_names_expr=["finger_joint"],
                effort_limit_sim=16.5,
                damping=0.02,
                stiffness=0.17
            ),
            "gripper_passive": ImplicitActuatorCfg(
                joint_names_expr=[".*inner_finger_joint.*"],
                effort_limit_sim=0.5,
                damping=0.001,
                stiffness=0.02
            ),
```

Also, wondering as to why you chose to disable self collisions? I believe that the gripper might rely on self collisions to move the internal joints.

**Comment by N3rv3rM1nd** [comment]
Thanks for your advice sincerely!

Your parameters really helped me a lot! Actually, I am quite new to this field and still don't fully grasp how these specific parameters impact the robot's behavior.

Previously, I temporarily disabled self-collisions to prevent the arm from oscillating wildly due to internal mesh clipping (a problem I later resolved by increasing the Newton engine's simulation frequency). I have now re-enabled self-collisions.

After running the simulation with your suggested parameters, the gripper still cannot be controlled properly, but I made a new discovery. As shown in the attached picture, when the gripper attempts to close, it does not do so symmetrically. Instead, both fingers move uniformly in the same direction, which eventually causes the left part to dislocate and detach from its pivot joint.

I will continue to investigate following this clue. If you have any insights or expert opinions on why this asymmetrical movement happens, I am all ears!

Additionally, I am still quite puzzled as to why the official USD assets I downloaded were configured with such extremely high stiffness values in the first place.

Anyway, thank you so much for your time and help!😊

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/8394e661-78ba-4aa2-a860-b0ca4e5fd0a2" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/81389622-ae4b-4b22-9dc5-a33f8d150861" />

**Comment by N3rv3rM1nd** [comment]
I want to sincerely thank you again for the parameter settings you provided! 🙏

With my previous settings, there was a constant slight oscillation that made it impossible for me to clearly observe the gripper's actual movements.

Following the clues, I just checked the gearing properties of the gripper and discovered something crazy: all the gearing values were magnified 100 times (e.g., 100 or -100)! This actually perfectly aligns with what happened when I first imported the gripper into the scene—its scale was also magnified by 100 times by default.

After restoring the gearing values back to their normal 1.0/-1.0 ratios, the gripper can finally open and close perfectly! I can now happily move forward with the rest of my experiments.😭

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/cd28962d-682e-4afe-9b8d-6e564d626403" />
